narwhal/consensus/src/consensus.rs

```diff
@@ -286,7 +287,9 @@ where
             state,
         };
 
-        spawn_monitored_task!(s.run())
+        spawn_monitored_task!(s.run()).tap(|_| {
+            info!("Consensus task shutdown");
+        })
```

Comment:
This does not do what you think it does. This will immediately return and print the log statement.

Reply:
If you want a statement to be printed when the task itself finishes, you'll want to directly instrument the future that's being spawned.

Reply:
oh my that's embarrassing - can't even think why I did it this way - I'll refactor  , thanks @bmwill !

Reply:
haha no worries, i've made similar mistakes before which is why i knew this wouldn't do what you want it to :)

Reply:
I would have made the same mistake! Thanks @bmwill